Add logging using python module - logging

sahildua2305 opened the pull request:
Using logging module, logging a message at INFO level

Fixes #36 

/cc @leplatrem @Natim

sahildua2305 commented:
@leplatrem I have added logging for only bucket's code as of now. Can you please have a look to confirm that I'm on right path?

Natim commented:
Can you add a test to make sure creating/updating/deleting/getting a bucket logs an info message?

You can use:

```
        with mock.patch('kinto-http.__innit__.logger') as mocked_logger:
            # Create a bucket
            mocked_logger.info.assert_called_with(
                "Create bucket 'accfe9ce-29ef-462e-aeea-ed8a5f0545d6' with data '{"foo: "bar"}'")
```

I think if we are to write the data there, we should probably indent it using `json.dumps`

sahildua2305 commented:
@Natim Should I add tests for the same in `test_client.py` or in a new test file for logging tests - 'test_logging.py`?

sahildua2305 commented:
@Natim Thanks for providing sample test. I have added tests for all logs that I have added so far. Please review it and point out any improvements. I will be adding more info logs.

sahildua2305 commented:
@Natim @leplatrem ready to go 👍

Natim commented on kinto_http/__init__.py:
```diff
@@ -213,6 +215,9 @@ def create_bucket(self, bucket=None, data=None, permissions=None,
         headers = DO_NOT_OVERWRITE if safe else None
         endpoint = self.get_endpoint('bucket', bucket=bucket)
+
+        logger.info("Create bucket %r with data %r" % (bucket, data))
```
Why don't you talk about the permission there while you do in the group creation method?

Natim replied:
I would probably log the data and permission only if it is present with another call to logger.
Maybe we can have an helper:

```
log_params(logger, "Create bucket", bucket=bucket, data=data, permissions=permissions, safe=safe, if_not_exists=if_not_exists)
```

leplatrem replied:
I don't think we should log data. Or if we do, we should do it in a second call with a `debug` level

sahildua2305 replied:
@leplatrem so should I completely remove data from `info` level messages and just keep the name of the object being created/updated/get-ed/deleted?

leplatrem replied:
Yes I think so

Natim commented on kinto_http/__init__.py:
```diff
@@ -222,6 +227,9 @@ def update_bucket(self, bucket=None, data=None, permissions=None,
         endpoint = self.get_endpoint('bucket', bucket=bucket)
         headers = self._get_cache_headers(safe, data, if_match)
+
+        logger.info("Update bucket %r with data %r" % (bucket, data))
```
```log_params(logger, "Update bucket", bucket=bucket, data=data, permissions=permissions, safe=safe, if_match=if_match, method=method)```

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -251,12 +262,18 @@ def delete_bucket(self, bucket=None, safe=True, if_match=None, if_exists=False):
         endpoint = self.get_endpoint('buckets')
         headers = self._get_cache_headers(safe, if_match=if_match)
+
+        logger.info("Delete buckets for if_match %r" % if_match)
```
`if_match` is superfluous here in my opinion

sahildua2305 replied:
Removed `if_match` from everywhere.

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -280,6 +297,11 @@ def create_group(self, group, bucket=None,
+
+        logger.info(
+          "Create group %r for bucket %r with data %r and permissions %r"
+          % (group, bucket, data, permissions))
```
c.f previous comment (data and permissions may not have to be always present)

leplatrem replied:
also I would say `group %r in bucket`, no?

sahildua2305 replied:
Yes that's better. Updated 👍

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -301,6 +323,11 @@ def update_group(self, group, data=None, bucket=None,
+
+        logger.info(
+          "Update group %r for bucket %r with data %r and permissions %r"
+          % (group, bucket, data, permissions))
```
ditto

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -330,12 +360,18 @@ def delete_group(self, group, bucket=None,
         endpoint = self.get_endpoint('groups', bucket=bucket)
         headers = self._get_cache_headers(safe, if_match=if_match)
+
+        logger.info("Delete groups for bucket %r for if_match %r" % (bucket, if_match))
```
ditto for `if_match`

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -359,6 +395,11 @@ def create_collection(self, collection=None, bucket=None,
                                      collection=collection)
+
+        logger.info(
+          "Create collection %r for bucket %r with data %r and permissions %r"
```
ditto for data and permissions.

Also, I realize that in the Kinto Admin UI we show something like that:  `Create collection blog/articles`. It makes it shorter to read than `%r in bucket %r`. What do you think?

sahildua2305 commented:
@Natim @leplatrem I have made changes as suggested by you. Please review 😄

leplatrem commented on kinto_http/__init__.py:
```diff
@@ -464,6 +516,11 @@ def create_record(self, data, id=None, collection=None, permissions=None,
                                      collection=collection)
+
+        logger.info(
+          "Create record in collection %r in bucket %r"
```
you could add `with id %r` here since we have it at hand

sahildua2305 replied:
Added 👍

sahildua2305 commented:
@Natim Please review again 🙂
